tests/test_pythonpackage.py

```diff
@@ -0,0 +1,133 @@
+
+    # The python binary needs to match our major version to be correct:
+    pyversion = subprocess.check_output([
+        pybin, "-c", "import sys; print(sys.version)"
+    ], stderr=subprocess.STDOUT)
+    try:
+        pyversion = pyversion.decode("utf-8", "replace")
+    except AttributeError:
+        pass
+    assert(pyversion.strip() == sys.version.strip())
+
+    # Helper function to run our function:
+    def run__get_system_python_executable(pybin):
+        # We want to see what _get_system_python_executable() does given
+        # a specific python, so we need to make it import it and run it:
+        cmd = [
+            pybin,
+            "-c",
+            "import importlib\n"
+            "import json\n"
+            "import os\n"
+            "import sys\n"
+            "sys.path = [os.path.dirname(sys.argv[1])] + sys.path\n"
+            "m = importlib.import_module(\n"
+            "    os.path.basename(sys.argv[1]).partition('.')[0]\n"
+            ")\n"
+            "print(m._get_system_python_executable())",
+            os.path.join(os.path.dirname(__file__), "..",
+                         "pythonforandroid", "pythonpackage.py"),
+        ]
+        # Actual call to python:
+        try:
+            result = subprocess.check_output(cmd, stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError as e:
+            raise RuntimeError("call failed, with output: " + str(e.output))
+        try:
+            return result.decode("utf-8", "replace").strip()
+        except AttributeError:
+            return result.strip()
+
+    # Check that in system-wide python, the system-wide python is returned:
+    try:
+        p1 = os.path.normpath(run__get_system_python_executable(pybin))
+        p2 = os.path.normpath(pybin)
+        assert(p1 == p2)
+    except RuntimeError as e:
+        if "pep517" in str(e.args):
+            # System python probably doesn't have pep517 available.
+            # Not much we can do in that case.
+            pass
+        else:
+            raise e
+
+    test_dir = tempfile.mkdtemp()
+    try:
+        # Check that in a virtualenv, the system-wide python is returned:
+        subprocess.check_output([
+            pybin, "-m", "virtualenv",
+            "--python=" + str(sys.executable),
+            "--",
+            os.path.join(test_dir, "virtualenv")
+        ])
+        subprocess.check_output([
+            os.path.join(test_dir, "virtualenv", "bin", "pip"),
+            "install", "-U", "pip"
+        ])
+        subprocess.check_output([
+            os.path.join(test_dir, "virtualenv", "bin", "pip"),
+            "install", "-U", "pep517"
+        ])
+        assert(os.path.normpath(run__get_system_python_executable(
+                                os.path.join(test_dir, "virtualenv",
+                                             "bin", "python"))) ==
+               os.path.normpath(pybin))
+        if int(sys.version.partition(".")[0]) >= 3:
```

Comment:
Conditional statements in a unit test with different code path kinda hurt me by default.
Not sure if it would make more sense or be more elegant to use `unittest.skipIf` for Python2. At the end it's probably the same because it using a conditional statement, but it makes it maybe more explicit

Reply:
Open for anything, I have to admit I'm not very familiar with them @ conditional statement/tests, so I just didn't think about it :grimacing:

Reply:
this is now a pytest skip thing! so I'll resolve this